virttest/utils_misc.py

```diff
@@ -2557,19 +2557,18 @@ def __get_unit_index(M):
             pass
         return 0
 
-    regex = r"(\d+)\s*(\w?)"
-    match = re.search(regex, value_str)
-    try:
```

Comment:
I don't know the problems that you encounter.
IMO, Here match the value string firstly is better than call isdigit() or isdecimal()?